Steps/BuildPlatform.yml

```diff
@@ -135,19 +139,29 @@ steps:
   condition: and(and(gt(variables.pkg_count, 0), succeeded()), eq(variables['Run'], true))
   timeoutInMinutes: ${{ parameters.run_timeout }}
 
-# Copy build logs to the artifact staging directory
-- template: CommonLogCopyAndPublish.yml
-  parameters:
-    artifacts_identifier: ${{ parameters.artifacts_identifier }}
+- ${{ if eq(parameters.publish_artifacts, true) }}:

```

Comment:
In general, I'm fine with the changes. But, there are two important changes in behavior that should be restored.

1. Before, each set of artifacts was published before the copy/publish for the next. The copy and publish tasks are sequenced so that if something went wrong highest priority artifacts (i.e. logs) to figure out what happened would be available. Given this is calling a template where the implementation is unknown, having publication right away was a step to help diagnose problems.
2. This is publishing the contents of the entire `$(Build.ArtifactStagingDirectory)` directory whereas previously individual paths were being published. In consumer pipelines, this might change what is published if contents were sitting in the artifacts directory and not explicitly published. That is an unnecessary change in behavior to accomplish this change.

To solve both of these, please put a publish step after each copy template matching the old publish step where the artifact specific directory is published.

Reply:
Restore the flow back and only introduce publish_artifacts parameter to control the execution of the artifact publish step.